Add some style invalidation tests for imperative slot API

rniwa opened the pull request:
#### c10d6b1a2919f1a4e8c882afe3e71008acf158bc
<pre>
Add some style invalidation tests for imperative slot API
<a href="https://bugs.webkit.org/show_bug.cgi?id=243923">https://bugs.webkit.org/show_bug.cgi?id=243923</a>

Reviewed by Simon Fraser.

Added style invalidation tests for the imperative slot API.

* LayoutTests/fast/shadow-dom/imperative-slot-update-1-expected.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-1.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-2-expected.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-2.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-3-expected.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-3.html: Added.

Canonical link: <a href="https://commits.webkit.org/253456@main">https://commits.webkit.org/253456@main</a>
</pre>

smfr commented on LayoutTests/fast/shadow-dom/imperative-slot-update-1.html:
```diff
@@ -0,0 +1,30 @@
+shadowRoot.appendChild(slot);
+slot.assign(document.querySelector('.fail'));
+
+requestAnimationFrame(() => {
```
Why does there need to be any asynchrony in this test? We always do a rendering update at the end of the test.

rniwa replied:
@smfr : oh, the point is to invalidate style after style had been updated.

rniwa committed:
Add some style invalidation tests for imperative slot API
https://bugs.webkit.org/show_bug.cgi?id=243923

Reviewed by Simon Fraser.

Added style invalidation tests for the imperative slot API.

* LayoutTests/fast/shadow-dom/imperative-slot-update-1-expected.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-1.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-2-expected.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-2.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-3-expected.html: Added.
* LayoutTests/fast/shadow-dom/imperative-slot-update-3.html: Added.

Canonical link: https://commits.webkit.org/253456@main
